### PR TITLE
Vanilla Zappa/Pyramid

### PR DESCRIPTION
**Pull request by Nico005**
Vanilla working Version of Zappa with Pyramid and MySQL

https://qj5ddm657a.execute-api.us-east-1.amazonaws.com/default

### CODE REVIEW

File: ansible/roles/base/tasks/main.yml [hunk 1]
```diff
@@ -1,14 +1,12 @@
-
```
Comment (oakensoul) [inline ansible/roles/base/tasks/main.yml:1]: A lot of strange spacing artifacts showed up in this file. Do you have any sense what happened? It looks like maybe you have your editor set to single space indents instead of two space indents? We should try to make sure we’re not constantly colliding. With the YAML files, I would prefer 2 spaces (not tabs). Let’s see if we can fix this before I accept the PR. If it’s my editor, I can fix it too, but either way we should double check to see what’s going on there.

File: ansible/roles/base/tasks/main.yml [hunk 2]
```diff
@@ -17,19 +15,24 @@
...
+       name: python-virtualenv
       become: yes
 
     - name: Install PIP requirements
+      become: yes
+      become_method: sudo -H
       pip:
         requirements: /vagrant/ansible/pip-requirements.txt
-        virtualenv: /home/vagrant/python3.6
+        virtualenv: /vagrant/python3.6
```
Comment (oakensoul) [inline ansible/roles/base/tasks/main.yml]: Why did you change this to the /vagrant/ folder?

Comment (oakensoul) [reply ansible/roles/base/tasks/main.yml]: Also, I see below, now all the files are in /python3.6/… I don’t think that’s what we want… That means when/if we ever change to 3.7 now we’re going to have a separate folder, etc. I’m not sure we want that. The root code should just be in /src/ or something like that.

File: ansible/vagrant.retry [hunk 3]
```diff
@@ -0,0 +1 @@
+127.0.0.1
```
Comment (oakensoul) [inline ansible/vagrant.retry]: We should add vagrant.retry to the .gitignore

File: python3.6/PyZAPI/development.ini [hunk 4]
```diff
@@ -0,0 +1,70 @@
+###
```
Comment (oakensoul) [inline python3.6/PyZAPI/development.ini]: We should not be committing our actual “development.ini” or “production.ini” files into the repository. We should craft a “developer.ini.example” if necessary. You don’t ever want to put in project specific or account specific items into GitHub (battlecry@) or the actual URL for your database in AWS. That exposes to the world where your database is sitting, and will provide an opportunity for someone to try and brute force themselves access.

**Review by oakensoul — CHANGES_REQUESTED** [review]
Also, I noticed you’re not following the Pull Request template that the project provides, you really should be doing that. There are sections in there that are very important… like What you’re adding, why, notes about features, how to test, etc.

Lastly, you should be updating the README with any instructions that are important for the user(s) of the project. For example, when you try to set this up, you need to edit the various config files to take whatever actions are necessary. If I were to try and clone this repository, what’s required for me to do to get it to work? That kind of stuff.

It’s also important that you try to clean up your git commit messages. There’s a LOT of code in here, so it’s important to try and break the commits up into pieces that make sense and are easy to digest for folks reviewing the code.

Overall though, it looks like it’s getting close!

File: python3.6/PyZAPI/zappa_settings.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "default": {
+        "app_function": "pyzapi.app.zappa_dev",
+        "aws_region": "us-east-1",
+        "profile_name": "default",
+        "s3_bucket": "zappa-7ghpa3yc7"
```
Comment (ajbosco) [inline python3.6/PyZAPI/zappa_settings.json]: missing `,`

**Review by ajbosco — COMMENTED** [review]
Can you also add `*.pyc` to the `.gitignore` file?

File: ansible/roles/base/tasks/main.yml (hunk 2 above)
Comment (Nico005) [reply ansible/roles/base/tasks/main.yml]: Hi Rob, Thanks for the honest and detailed comments.
Reason i changed the virtualenv to /vagrant/ was to sync the files back to my local so i can commit.
Also Zappa needs to run inside the virtualenv, hence the project was put inside the python3.6 folder or should we create the virtualenv outside the python3.6 folder?

File: ansible/vagrant.retry (hunk 3 above)
Comment (Nico005) [reply ansible/vagrant.retry]: Great, will add the *.pyc and retry file to gitignore for future commits

File: python3.6/PyZAPI/development.ini (hunk 4 above)
Comment (Nico005) [reply python3.6/PyZAPI/development.ini]: Agreed, we do the same here with our config files.
Will make the changes, and add the ini files to gitignore

File: python3.6/PyZAPI/zappa_settings.json (hunk 5 above)
Comment (Nico005) [reply python3.6/PyZAPI/zappa_settings.json]: Comma added, thanks

File: ansible/roles/base/tasks/main.yml (hunk 1 above)
Comment (Nico005) [reply ansible/roles/base/tasks/main.yml:1]: Hi Rob, i use vim/vi or even notepad lol to edit my files at the mooment.
What editor are you using? Ill download and use the same one to be consistent.

Comment (oakensoul) [reply ansible/roles/base/tasks/main.yml:1]: Personally, I use PyCharm from JetBrains (relatively inexpensive). Some folks also like Atom (Free), but that may require some customization.
